app/services/recipe_generator_service.rb

```diff
@@ -60,6 +70,8 @@ def create_recipe(response)
     parsed_response = response.is_a?(String) ? JSON.parse(response) : response
     content = JSON.parse(parsed_response.dig('choices', 0, 'message', 'content'))
     # create recipe here
+    recipe = @user.recipes.build(name: content['name'], description: content['description'], ingredients: @message)
```

Comment:
Build does not save the object in the DB, maybe that's why you are doing `recipe` after creation?
```suggestion
    recipe = @user.recipes.create(name: content['name'], description: content['description'], ingredients: @message)
```